Feature/146683087/get jwt on login

codejockie opened the pull request:
Pull Request Convention

#146683087 Users should receive a JWT on successful login
#146683089 Users should receive a JWT on successful signup

    #### What does this PR do?
    #### This PR changes the authentication method from session auth to JWT auth

    #### Description of Tasks to be completed
    #### Users once authenticated via the login api should be returned a JWT to be required on subsequent call to all other endpoints that require authentication
    #### Users once authenticated via the signup api should be returned a JWT to be required on subsequent call to all other endpoints that require authentication

    #### How should this be manually tested?
    #### N/A

    #### Any background context you want to provide?
    #### No

    #### What are the relevant pivotal tracker stories?
    #### Users should receive a JWT on successful login
    #### Users should receive a JWT on successful signup

    #### Screenshots (if appropriate)
    #### None

    #### Questions:
    #### None

Hector101 committed:
chore(install-eslint): install eslint to project

Start #146683069

Hector101 committed:
chore(eslint-setup): configure eslint to project,
add eslint dependencies to project

Finishes #146683069

Hector101 committed:
chore(eslint-setup): add require-jsdocs rule to enforce  user to add a js-docs on class methods and functions

Finishes #146683069

Commit:
chore: app-stabilisation

created sample .env file
add dotenv module to server.js to load environment variables

finishes #146687435

temilaj committed:
Merge pull request #4 from andela/chore/146687435/stabilise-base-application

#146687435 stabilise base application

Commit:
chore: app-stabilisation

change version of connect mongo in package.json
change version of node and npm in package.json

finishes #146683067

Commit:
chore(travisci): integrate travisci with readme badge

pull the most recent version of the code and open another pull request

Finishes #146683077

Commit:
chore(travis): integrate travisci with readme badge

update travisci

Finishes #146683077

Hector101 committed:
chore(eslint-setup): configure eslint to project
add eslint dependencies to project

Finishes #146683069

Commit:
chore(travisci): integrate Travisci with readme badge

delete article model because it doesn't have a schema registered for it

Finishes #146683077

Commit:
chore(travisci): integrate travisci with readme file

comment out a test that needs a database connection to see if travis will flag another error

Finishes #146683077

Commit:
chore(travisci): integrate travisci with readme file

uncomment out the failing test to rebuild

Finishes #146683077

temilaj committed:
Merge pull request #8 from andela/chore/146683069/setup-eslint

#146683069 setup eslint configuration

Commit:
chore(travisci): integrate travisci with readme file

revert version of node and npm to the initial ones

Finishes #146683077

Commit:
chore(travisci): integrate travisci with readme file

update the version of node and npm in package.json

Finishes #146683077

Commit:
chore(travisci): integrate travisci with readme file

update version of npm and node in package.json

Finishes #146683077

Commit:
chore(travisci): integrate travisci with readme file

update version of npm and node in package.json

Finishes #146683077

Commit:
chore(travisci): integrate travisci with readme badge

update version of npm and node in package.json

Finishes #146683077

Commit:
chore(travisci): integrate travisci with readme file

change the version of nodejs in .travis.yml

Finishes #146683077

Commit:
chore(travisci): integrate travisci with readme file

update version of node in travis.yml

Finishes #146683077

Commit:
chore(travisci): integrate travisci with readme badge

Modify node_js version in .travis.yml

Finishes #146683077

temilaj committed:
Merge pull request #9 from andela/chore/146683077/integrate-travisci-with-readme-badge

#146683077 Integrate TravisCI with readme badge

baasbank committed:
Adds coveralls badge to README

baasbank committed:
Creates .coverall.yml file

baasbank committed:
chore(coveralls): resolve merge conflict

Start #146683085

baasbank committed:
chore(continuous integration): integrate coveralls ci

 Fixes #146683085

baasbank committed:
chore(coverage-report): integrate coveralls

Commit:
chore(gulp): convert from grunt to gulp taskrunner

Create gulpfile.js and install dependencies, Delete gruntfile.js and remove dependencies, Ignore bower_components, modify scripts to use gulp in package.json.

Finishes #146683083

Commit:
chore(gulp): convert from grunt to gulp task runner

Ignore .env file

Finishes #146683083

Commit:
chore(gulp): convert from grunt to gulp task runner

modify gulpfile.js

Finishes #146683083

baasbank committed:
chore(coverage-report): integrate coveralls

tonerdo committed:
Merge pull request #17 from andela/chore/146683085/Integrate-Coveralls-code-coverage-service

Chore/146683085/integrate coveralls code coverage service

Commit:
chore(gulp): convert from grunt to gulp task runner

fix merge conflicts

Finishes #146683083

Commit:
chore(gulp): convert from grunt to gulp task runner

fix merge conflicts

Finishes #146683083

Commit:
chore(gulp): convert from grunt to gulp task runner

fix merge conflicts

Finishes #146683083

Commit:
chore(gulp): convert from grunt to gulp task runner

fix merge conflicts

Finishes #146683083

Commit:
chore: app-stabilisation

change mongo url
change express connection in server.js

Commit:
chore: app-stabilisation

move grunt modules from dev dependencies to main app dependency

Commit:
chore: app-stabilisation

Fix merge conflict with staging

Commit:
chore: app-stabilisation

add jade to engines

Commit:
chore: app-stabilisation

add bower install to script in package.json

tonerdo committed:
Merge pull request #14 from andela/chore/146683083/convert-from-grunt-to-gulp-taskrunner

#146683083 Convert from Grunt to Gulp taskrunner

Commit:
chore: app-stabilisation

fix dependency issues

finishes #146683067

Commit:
chore: app-stabilisation

fix merge conflict in package.json and .gitignore file

finishes #146683067

Commit:
chore: app-stabilisation

remove sample.env and merge conflict markers in .gitignore

finishes #146683067

Commit:
chore: app-stabilisation

added default value for db

finishes #146683067

Commit:
chore: app-stabilisation

change mongodb default connection url

finishes #146683067

tonerdo committed:
Merge pull request #19 from andela/chore/146687435/stabilise-base-application

Chore/146687435/stabilise base application

codejockie committed:
feature(flight): install jsonwebtoken package

this package include JWT functions

Start #146683089

codejockie committed:
feature(authentication): user registration

a user receives a token on sign up

Finishes #146683089

codejockie committed:
feature(authentication): user login

a user receives a token on login.
add: public/js/controllers/login.js
add login function in auth.js
add login.js to foot.jade
add login route to routes.js
add logout function to public/js/controllers/index.js
add token check in directives.js
add logout handler to index.html
add login handlers to signin.html

Finishes #146683087

tonerdo commented on public/js/directives.js:
```diff
@@ -71,9 +71,13 @@ angular.module('mean.directives', [])
-        }
+        // if (scope.$$childHead.global.authenticated === true) {
+        //   scope.showOptions = false;
+        // }
```
Don't commit commented out code

codejockie replied:
should I close the PR?

tonerdo replied:
@codejockie please don't ask me this question. I've talked with you guys about this before

codejockie replied:
I know what to do. Thank you.

tonerdo commented on server.js:
```diff
@@ -1,6 +1,7 @@
 /**
  * Module dependencies.
  */
+require('dotenv').config();
```
Why? You're not using an `.env` file

codejockie replied:
I didn't add that require statement, it was part of the original codebase. for the commented code, I'd fix that.

codejockie committed:
feature(authentication): user login

a user receives a token on login.
removed commented out code from directives.js

Finishes #146683087

codejockie commented on server.js:
```diff
@@ -1,6 +1,7 @@
 /**
  * Module dependencies.
  */
+require('dotenv').config();
```
Done.
